Makefile

```diff
@@ -252,6 +252,30 @@ $(STABLE_BIN)/$(PROTOC_VERSION_BIN): | $(STABLE_BIN)
 	$Q unzip -q $(STABLE_BIN)/protoc.zip -d $(PROTOC_UNZIP_DIR)
 	$Q cp $(PROTOC_UNZIP_DIR)/bin/protoc $@
 
+# checks that the idl submodule points to a commit on master, and that it matches the go module (which must be a pseudo version).
+# this is only used in an explicit CI step, because it's expected to fail when developing.
+#
+# `git ls-tree HEAD idls` is selected because this only cares about the committed/checked-out target,
+# not whatever the current status is, because only the committed value will exist for others.
+#
+# and last but not least: this avoids using `go` to make this check take only a couple seconds in CI,
+# so the whole docker container doesn't have to be prepared.
+.idl-status:
+	branches="$$(git submodule foreach git branch master --contains HEAD)"; \
```

Comment:
does this work? I get some spurious output: 
```
git submodule foreach git branch master --contains HEAD | grep id
Entering 'idls'
```

Reply:
oh, I didn't read the line below correctly, you're checking for the presence of the master branch (whcih I currently don't have). ok, np

Reply:
yea.  "working" is
```
Entering 'idls'
 master
```
and "not" is
```
Entering 'idls'
```

`git submodule foreach` is just a bit easier than cd-ing around and dealing with those details, since I couldn't find a way to tell git to do "in this submodule only, do X".